lib/puppet/provider/gnupg_key/gnupg.rb

```diff
@@ -73,7 +79,7 @@ def add_key_from_key_content
     path = create_temporary_file(user_id, resource[:key_content])
     command = "gpg --import #{path}"
     begin
-      output = Puppet::Util::Execution.execute(command, :uid => user_id, :failonfail => true)
+      `#{command}`
```

Comment:
That raises an exception and halts the script despite everything being fine, literally output of gpg includes no errors, exception gets thrown anyway. This way puppet only errors if there's... you know, an error.